renderer/components/ui/DisclosureSection.tsx

```diff
@@ -1,41 +1,52 @@
 import clsx from "clsx";
 import { ChevronDownIcon } from "@heroicons/react/24/solid";
-import React from "react";
+import { Disclosure } from "@headlessui/react";
+import React, { ReactNode } from "react";
 
-export default function DisclosureSection(props) {
+type DisclosureSectionProps = {
+  toggleFunction: () => void;
+  isOpen: boolean;
+  title: string;
+  children: ReactNode;
+};
+
+export default function DisclosureSection(props: DisclosureSectionProps) {
   return (
     <div className="lg:absolute mt-6 w-full">
-      <div
-        className={clsx(
-          "h-16 px-4 py-5 bg-white shadow overflow-hidden transition-all ease-linear duration-300 sm:rounded-lg sm:px-6",
-          {
-            "h-80 overflow-y-auto ": props.isOpen,
-          }
-        )}
-      >
-        <div
-          className="flex justify-between  cursor-pointer"
-          onClick={props.toggleFunction}
-        >
-          <h2
-            id="manual-input-title"
-            className="text-lg font-medium text-gray-900"
-          >
-            {props.title}
-          </h2>
-          <button
+      <Disclosure>
+        {({ open }) => (
+          <div
             className={clsx(
-              "transform transition-transform ease-linear duration-300",
+              "max-h-16 px-4 py-4 bg-white shadow overflow-hidden transition-all ease-linear duration-300 sm:rounded-lg sm:px-6",
               {
-                "rotate-180": props.isOpen,
+                "max-h-80 overflow-y-auto ": open,
               }
             )}
           >
-            <ChevronDownIcon className="w-6 h-7" aria-hidden="true" />
-          </button>
-        </div>
-        {props.children}
-      </div>
+            <Disclosure.Button className=" w-full">
+              <div className="flex justify-between cursor-pointer">
+                <h2
+                  id="manual-input-title"
+                  className="text-lg font-medium text-gray-900"
+                >
+                  {props.title}
+                </h2>
+                <button
+                  className={clsx(
+                    "transform transition-transform ease-linear duration-300",
+                    {
+                      "rotate-180": open,
+                    }
+                  )}
+                >
+                  <ChevronDownIcon className="w-6 h-7" aria-hidden="true" />
+                </button>
```

Comment:
button inside button trigger warning in console about "validateDOMNesting". I think you can use only icon in this case, because disclosure opens on roots div click, it is enough.